app/code/community/Algolia/Algoliasearch/Helper/Entity/Producthelper.php

```diff
@@ -512,102 +521,100 @@ protected function handlePrice(Mage_Catalog_Model_Product &$product, $sub_produc
-
         if ($customer_groups_enabled) {
             $groups = Mage::getModel('customer/group')->getCollection();
+        } else {
+            $groups = array();
+        }
+        $special_price = $product->getSpecialPrice();
+        $special_from_date = $product->getSpecialFromDate();
+        $special_to_date = $product->getSpecialToDate();
+
+        // There is a special_price and either no to date or a to date in the future
+        // start date is irrelevant
+        if ($special_price && Mage::app()->getLocale()->isStoreDateInInterval($store, null, $special_to_date)) {
+            $canSpecialBeValid = true;
+            $special_price += $weeeTaxAmount;
         }
-
-        /** @var Mage_Tax_Helper_Data $taxHelper */
-        $taxHelper = Mage::helper('tax');
-
-        /** @var Mage_Directory_Helper_Data $directoryHelper */
-        $directoryHelper = Mage::helper('directory');
 
         foreach ($fields as $field => $with_tax) {
             $customData[$field] = array();
+            $field_price = (double) $taxHelper->getPrice($product, $product->getPrice() + $weeeTaxAmount, $with_tax, null, null, null, $store, null);
+            if ($canSpecialBeValid) {
+                $field_special_price = (double) $taxHelper->getPrice($product, $special_price, $with_tax, null, null, null, $store, null);
+            }
 
             foreach ($currencies as $currency_code) {
-                $customData[$field][$currency_code] = array();
+                $cdfc = array();
 
-                $price = (double) $taxHelper->getPrice($product, $product->getPrice(), $with_tax, null, null, null, $product->getStore(), null);
-                $price = $directoryHelper->currencyConvert($price, $baseCurrencyCode, $currency_code);
-                $price += $weeeTaxAmount;
+                $price = $directoryHelper->currencyConvert($field_price + $weeeTaxAmount, $baseCurrencyCode, $currency_code);
 
-                $customData[$field][$currency_code]['default'] = $price;
-                $customData[$field][$currency_code]['default_formated'] = $this->formatPrice($price, false, $currency_code);
+                $cdfc['default'] = $price;
+                $cdfc['default_formated'] = $this->formatPrice($price, false, $currency_code);
 
-                $special_price = (double) $taxHelper->getPrice($product, $product->getFinalPrice(), $with_tax, null, null, null, $product->getStore(), null);
-                $special_price = $directoryHelper->currencyConvert($special_price, $baseCurrencyCode, $currency_code);
-                $special_price += $weeeTaxAmount;
+                if ($canSpecialBeValid) {
+                    $special_price = $directoryHelper->currencyConvert($field_special_price + $weeeTaxAmount, $baseCurrencyCode, $currency_code);
 
+                    // this needs moving up a few levels
+                    $cdfc['special_from_date'] = Mage::app()->getLocale()->storeDate($store, $special_from_date)->getTimestamp();
```

Comment:
I'm using ->storeDate here because strtotime wasn't time zone aware, and configuration.phtml was using storeDate so I was get off by 3600 hours because of DST